container-search/src/main/java/com/yahoo/search/query/QueryTree.java

```diff
@@ -109,6 +109,19 @@ public boolean isEmpty() {
 
     // -------------- Facade
 
+    /**
+     * Modifies this query to become the current query RANK with the given item.
+     *
+     * @return the resulting root item in this
+     */
+    public Item rankWith(Item item) {
```

Comment:
Wouldn't it be more logical with "withRank(item)"?

But I think it would be best to just name it "rank" to match with the "and" methods which are also in this facade.